cli_tools/common/utils/logging/service/log_entry.go

```diff
@@ -171,27 +184,23 @@ type OutputInfo struct {
 	FailureMessageWithoutPrivacyInfo string `json:"failure_message_without_privacy_info,omitempty"`
 }
 
-func (l *Logger) updateParams(projectPointer *string) {
+func (l *Logger) updateParams(params map[string]string) {
 	l.mutex.Lock()
 	defer l.mutex.Unlock()
 
-	if projectPointer == nil {
+	l.updateProject(params)
+}
+
+func (l *Logger) updateProject(params map[string]string) {
```

Comment:
since params are predefined, a struct is more appropriate instead of a map. This way, if some values in the future are not strings, we will be able to support them

Reply:
I chose not to use a struct because the implementation (exporter, for example) shouldn't ref to the structs defined specifically for logging. The logging framework only allows main function to interact with logging.service package. The import, export and ovf_importer knows nothing about logging and can be executed separately.

Reply:
See my other comment about the location of some of the structs/interfaces.

Also, CLI tools already refer to logging/service so they are aware of being run within logging framework. E.g. they import
"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/logging/service"

and implement InitParamLog

Reply:
Only main function ref to logging/service, but not the core exporter/importer/ovfImproter. Just like we have daisy and daisy cli. That is what I am trying to achieve.

Reply:
Then have this struct in a package common to all importer/exporter or CLI tools in general. 

By using a map, we're not achieving encapsulation or separation of concerns because importer/exporter still have to be aware they need to return some parameters. We only loose type safety with the current implementation.

Reply:
I read again about your comment. My design was: RunCliToolWithLogging ask for a returned map, and consume it in its own way. Thus, we can change the logging implementation safely. So I tried to avoid a fixed struct from service package. Your suggestion sounds reasonable: a struct from common package doesn't force tools to understand how logging is implemented. Let me do it.

Overall, Eric prefer a different pattern of architect. Let me finish my implementation, and you can discuss more which one makes more sense when I am oof...